invokeai/app/invocations/generate.py

```diff
@@ -41,8 +43,15 @@ class TextToImageInvocation(BaseInvocation):
 
     # TODO: pass this an emitter method or something? or a session for dispatching?
     def dispatch_progress(
-        self, context: InvocationContext, sample: Any = None, step: int = 0
+        self, context: InvocationContext, intermediate_state: PipelineIntermediateState
     ) -> None:
+        step = intermediate_state.step
+        # if intermediate_state.predicted_original is not None:
```

Comment:
i'm guessing this commented-out text should be deleted?

Reply:
I think (correct me if I'm wrong) `app` is the new web server that goes with nodes.

The callback here was written to take `sample`, but it doesn't _use_ it yet, so I wanted to leave a hint about how to get it.

Trying abide by our current "if you add features you should add them to nodes" guidelines, but fumbling because I haven't seen nodes frontend yet.

Reply:
I've modified and activated the code in question, but do not have a way of testing it at the moment. The code will dispatch an `emit_generator_progress()` event containing the decoded progress image as a `PIL.Image` object.